saml/src/main/java/com/linecorp/armeria/server/saml/KeyStoreCredentialResolverBuilder.java

```diff
@@ -56,15 +93,33 @@ public KeyStoreCredentialResolverBuilder addKeyPassword(String name, String pass
         return this;
     }
 
+    /**
+     * Adds all key names and their passwords which are specified by the {@code keyPasswords}.
+     */
+    public KeyStoreCredentialResolverBuilder addKeyPasswords(Map<String, String> keyPasswords) {
+        requireNonNull(keyPasswords, "keyPasswords");
+        keyPasswords.forEach(this::addKeyPassword);
+        return this;
+    }
+
     /**
      * Creates a new {@link KeyStoreCredentialResolver}.
      */
     public CredentialResolver build() throws IOException, GeneralSecurityException {
-        final KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
-        try (InputStream is = KeyStoreCredentialResolverBuilder.class.getClassLoader()
-                                                                     .getResourceAsStream(path)) {
-            ks.load(is, keystorePassword.toCharArray());
+        final KeyStore ks = KeyStore.getInstance(type);
+        try (InputStream is = open()) {
+            ks.load(is, password != null ? password.toCharArray() : null);
         }
         return new KeyStoreCredentialResolver(ks, keyPasswords);
     }
+
+    private InputStream open() throws IOException {
+        if (file != null) {
+            return new FileInputStream(file);
+        }
+
+        assert classLoader != null : "classLoader";
```

Comment:
Looking at these extra null checks, I don't know if it's a better pattern but came up with this. Any thoughts?

```java

private final Supplier<InputStream> keystoreStream;

public KeyStoreCredentialResolverBuilder(File file) {
  requireNonNull(file, "file);
  keystoreStream = () -> new FileInputStream(file);
}

public KeyStoreCredentialResolverBuilder(String, ClassLoader) {
  keystoreStream = () -> classLoader.getResourceAsStream(resourcePath);
}
```

Actually, after writing it though, I guess the annoying `IOException` makes it non-trivial 😿

Reply:
That's a good question. It may be useful to provide an abstraction for this:

```java
public class FileOrResource { ... }

public KeyStoreCredentialResolverBuilder(File file) {
  requireNonNull(file, "file);
  keystoreResource = new FileOrResource(file);
}

public KeyStoreCredentialResolverBuilder(String, ClassLoader) {
  keystoreResource = new FileOrResource(resourcePath, classLoader);
}
...

keysotreResource.newInputStream();
```

Reply:
@anuraaga @trustin Both sound good to me. But it seems that it's time to release. So, let me make a fix for the next version. Or, @anuraaga , you can take it if you are interested. :-)